### DISCUSSION TITLE
Digital Meter Diehl Hydrus 2

### DISCUSSION THREAD

**Discussion by tschuster89** [jomjol/AI-on-the-edge-device#2630] [Q&A]
Hey guys,

I've got a new digital meter some days ago. My idea is to use the esp-cam the same way than before and read the digital digits out of the picture. The meter/display starts if I open the lid (looks like the is a photo diod inside) If I leave the lid open, display shut off some minutes later. If I use a flashlight in front of the diod the display comes back up so my idea was to use the flash led from esp to wake up the display and after that take a picture like it was done before. May some of you has any idea how I can do this.

Please excuse my poor english. 
Thanks a lot, Tim

**Comment by caco3** [comment]
You could try to use https://jomjol.github.io/AI-on-the-edge-device-docs/Parameters/#parameter-waitbeforetakingpicture and switch the Flash light on some seconds before taking the picture.

**Comment by Krusty1977** [comment]
Has anybody out there already finished this project successfully? @jomjol do you know if anybody managed this already?

**Reply by caco3** [reply]
Did you try my suggestion?

Also be aware that such meter is not made to be read periodically! The battery will drain much faster than designed! Unless you are able to replace the battery by yourself, I would only use a very low interval!

**Reply by Krusty1977** [reply]
No, since today I read a very old water meter, but this would be the next step. So I asked if somebody has experience

**Comment by adx-git** [comment]
I have it working since a few months now.
I am using a delay of 5 seconds between LED turn on and taking the picture.
The picture is taken every three minutes.

I understand that this will heavily drain the battery of the watermeter, unfortunately this is the only possibility I have to read the water consumption. I requested the key to decrypt the periodically sent data, but my local supplier won't give it to me because of data privacy reasons.
At least during summer time I will keep it running, since I am afraid my automatic garden watering could be malfunctioning (which already happened once)

**Reply by caco3** [reply]
It also might be worth to read this (German) forum thread: https://forum.iobroker.net/topic/51163/wasserz%25C3%25A4hler-hydrus-2-0-von-diehl/114 (check the comments before and after)

**Comment by dabepp** [comment]
I have the same meter and my issue is more to get a clear picture without reflections. How did you all manage to get a proper reading?
<img width="403" height="123" alt="image" src="https://github.com/user-attachments/assets/2256ea34-6237-421d-9473-af4f3e5f41d3" />

**Reply by adx-git** [reply]
I got some reflection too, but I guess you need to adjust the focus first.

<img width="418" height="141" alt="image" src="https://github.com/user-attachments/assets/fd0b655e-8ef9-4dcf-ac12-b0d09c81f2b6" />
The reflection wasn't an issue for me.

**Reply by dabepp** [reply]
The only digit that bugs me is the 2nd, it is interpreted as 0 not as 7 all the time...

**Reply by adx-git** [reply]
Did you try improving the focus? This made a huge difference for me. 
The model, which I am using for digital processing, is "dig-class11_1910_s2_q.tflite".
There is no specific reason why I am using this one, but since it is working I am not changing it ...

**Reply by dabepp** [reply]
I finally figured it out. I built a top and bottom frame, put some cardboard with white paper covered faces inside in them, cut two holes where I put LED‘s with a reflective g 45 ° angle in it and here we are, no reflections. Sharp enough to get recognised w/o any issues!
![IMG_0560](https://github.com/user-attachments/assets/ba1b92b7-2a1d-4b1a-b6ff-898ce56da7c7)
![IMG_0554](https://github.com/user-attachments/assets/1fb08c85-ae74-4d34-bf54-50a05a664752)
